File: Medium/1481. Least Number of Unique Integers after K Removals/sol.cpp
class Solution {
public:
    int findLeastNumOfUniqueInts(vector<int>& arr, int k) {
        unordered_map<int, int> mp;
        deque<int> freq;

        for(auto & i : arr) mp[i]++;

        for(auto & i : mp) freq.emplace_back(i.second);
        sort(freq.begin(), freq.end());

        for(auto & i : freq) {
            if(k >= i) k -= i, freq.pop_front();
            else break;
        }

        return (int) freq.size();
    }
};

File: Medium/1481. Least Number of Unique Integers after K Removals/sol.py
class Solution(object):
    def findLeastNumOfUniqueInts(self, arr, k):
        mp = {}
        freq = []

        for i in arr:
            if i in mp: mp[i] += 1
            else: mp[i] = 1

        for i in mp: freq.append(mp[i])

        freq.sort()

        while freq:
            i = freq[0]
            if k >= i:
                k -= i
                freq.pop(0)
            else:
                break

        return len(freq)
       